js/packages/ufg-client/src/resources/process-resources.ts

```diff
@@ -252,4 +252,26 @@ export function makeProcessResources({
       return []
     }
   }
+
+  // Note: this function is needed since the contract with async cache is to return a serializeable object.
+  // So we need to peel off `ready` which is a promise, but still use it as the return value from the callback.
+  function getCachedResource(
+    key: string,
+    callback: () => Promise<CacheableKnownResource & {ready?: boolean | Promise<boolean>}>,
+  ): Promise<CacheableKnownResource> {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const resourceFromCache = await asyncCache!.getCachedResource(key, async () => {
+          const ret = await callback()
+          resolve(ret)
+          const retWithoutReady = {...ret}
+          delete retWithoutReady.ready
```

Comment:
```suggestion
          const {ready, ...retWithoutReady} = ret
```
i think that spread and delete is a bit over kill...

Reply:
yes, thanks!